physical/redis.go

```diff
@@ -0,0 +1,314 @@
+package physical
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/garyburd/redigo/redis"
+)
+
+// RedisBackend is a physical backend that stores data at a specific prefix
+// within Redis.  This is not especially useful as an ordinary
+// secret-storage backend, unless you set up Redis persistence.
+//
+// But this backend offers a relatively simple way to set up a "medium
+// availability" Vault cluster, where you use S3 or another backend for
+// storage, and a single Redis server as a "ha_backend" to handle leader
+// elections.  If the Redis server fails, then all vault servers should go
+// to standby until Redis returns.
+//
+// This is potentially useful because Redis servers are ubiquitous and
+// surprisingly reliable, and because not everybody wants to set up and
+// administer either Consul or etcd for a single leader lock.  Also,
+// high-quality hosted and managed Redis services are widely available.
+type RedisBackend struct {
+	path   string
+	pool   *redis.Pool
+	logger *log.Logger
+}
+
+func newRedisBackend(conf map[string]string, logger *log.Logger) (Backend, error) {
+	// Get the Redis path from the configuration.
+	path, ok := conf["path"]
+	if !ok {
+		path = "vault"
+	}
+
+	// Get the Redis URL from our environment.
+	url := os.Getenv("REDIS_URL")
+	if url == "" {
+		url = "redis://127.0.0.1:6379"
+	}
+	addr := strings.TrimPrefix(url, "redis://")
+
+	// Create a Redis connection pool so that we can access Redis from
+	// multiple goroutines in parallel.
+	pool := &redis.Pool{
+		MaxIdle:     2,
+		IdleTimeout: 300 * time.Second,
+		Dial: func() (redis.Conn, error) {
+			conn, err := redis.Dial("tcp", addr)
```

Comment:
Yes, that appears to be the case. If I've misremembered, I'll find out shortly (we'll test this with a tiny standalone Redis).